### ISSUE TITLE
VAE Reconstruction Inference Issue

### ISSUE THREAD

**Issue by kangyiyang** [nv-tlabs/XCube#47]
# VAE Reconstruction Inference Issue Report

## Problem Description
During my attempt to perform variational autoencoder (VAE) reconstruction inference, I encountered some technical difficulties. Here are my implementation steps and observed issues:

## Implementation Steps
1. Executed `shapenet_example.py` to generate pkl files containing voxel and normal data of meshes
2. Performed mesh inference using pre-trained weights from `fine_vae` and `nksr_refine` with the following code snippet:

```python
input_data = torch.load(os.path.join(voxel_dir, voxel_id))
input_points = input_data['points']
input_normals = input_data['normals'].jdata

save_path = os.path.join(target_dir, voxel_id.split(".")[0] + ".obj")

data = {
    DS.SHAPE_NAME: voxel_id.split(".")[0],
    DS.TARGET_NORMAL: input_normals.cuda(),
    DS.INPUT_PC: input_points,
    DS.GT_DENSE_PC: input_points,
    DS.GT_DENSE_NORMAL: input_normals.cuda()
}

out = {}
res, output_x = net_model_c(data, out)  # Modified autoencoder.py to return output_x

for batch_idx in range(output_x.grid.grid_count):
    pd_grid = output_x.grid[batch_idx]
    pd_normal = res['normal_features'][-1].feature[batch_idx].jdata

    with torch.no_grad():
        # Option 1: Using VAE-processed input
        field = reconstructor.forward({'in_grid': pd_grid, 'in_normal': pd_normal})
        
        # Option 2: Using original input
        # field = reconstructor.forward({'in_grid': input_points.to('cuda'), 'in_normal': input_normals.cuda()})
    
    field = field['neural_udf']
    
    mesh = field.extract_dual_mesh(grid_upsample=2)
    trimesh.Trimesh(
        vertices=mesh.v.cpu().detach().numpy(),
        faces=mesh.f.cpu().detach().numpy()
    ).export(save_path)

torch.cuda.empty_cache()
```
When using VAE-processed input (pd_grid and pd_normal), the reconstruction results show: 

![Image](https://github.com/user-attachments/assets/17a07de7-1350-44f7-9693-83cea10b29f8)

When using original input (input_points and input_normals), the reconstruction results show:

![Image](https://github.com/user-attachments/assets/6b1214d7-617d-4804-a57a-c441c32ef0f0)

**Comment by kangyiyang** [comment]
It seems there is an issue with shapenet_sample.py. When the grid size is set to 512, the target voxel size should be set to 0.001953125 under normalized conditions.

**Comment by ChnWuyue** [comment]
hello, could u tell me how do u fix the bug?

**Comment by kangyiyang** [comment]
You can try changing the line 89 in shapenet_example.py from target_voxel_size = 0.0025 to target_voxel_size = 0.001953125.

**Comment by ChnWuyue** [comment]
> You can try changing the line 89 in shapenet_example.py from target_voxel_size = 0.0025 to target_voxel_size = 0.001953125.

thanks for your suggestion, best wish for u.
